src/RTTComms.cpp

```diff
@@ -537,12 +540,20 @@ namespace BrainCloud
             _eventQueueMutex.lock();
             _callbackEventQueue.push_back(RTTCallback(RTTCallbackType::ConnectSuccess));
             _eventQueueMutex.unlock();
+            }
+            else if (operation == "DISCONNECT")
+            {
+                _disconnectedWithReason = true; // RESET THIS VALUE IN CLEANUP/connect()/ETC. Very error prone addition. be careful, have me review
+                _disconnectReasonMessage = json["data"]["status_message"].asString(); // Check log figure out what the message is
+                _disconnectReasonCode = json["data"]["reason_code"].asInt();
+                // Do not callback, do not take action. Add this message/reason once the socket dies.
```

Comment:
You forgot this part. Once the socket disconnect, use this reason code/message if `_disconnectedWithReason` is true